File: day1/day1.py
import pandas as pd

data = pd.read_csv("day1/input.csv", header = None)[0]

#PART 1

counter = 0
ln = len(data) - 1

for i in range(ln):
    if data[i + 1] > data[i]:
        counter += 1

print(counter)

#PART 2

counter = 0
ln = len(data) - 3

for i in range(ln):

    a = sum(data[i:i+3])
    b = sum(data[i+1:i+4])

    if b > a:
        counter +=1

print(counter)

File: day1/day1.jl
using Revise
using CSV
using DataFrames

df = CSV.read("day1/input.csv", DataFrame, header = 0)

col = df[:, 1]

## part 1

counter = 0
ln = length(col) - 1

for i in 1:ln
    if col[i] < col[i+1]
        global counter += 1
    end
end

counter

## part 2

counter = 0
ln = length(col) - 3

for i in 1:ln
    a = sum(col[i:i+2])
    b = sum(col[i+1:i+3])

    if b > a
        global counter += 1
    end
end

counter